substrate/frame/glutton/src/lib.rs

```diff
@@ -208,6 +208,51 @@ pub mod pallet {
 		}
 	}
 
+	/// Dummy error type.
+	#[derive(codec::Encode, sp_runtime::RuntimeDebug)]
+	#[cfg_attr(feature = "std", derive(codec::Decode))]
+	pub enum InherentError {
+		/// Fatal error.
+		Fatal,
+	}
+
+	impl frame_support::inherent::IsFatalError for InherentError {
+		fn is_fatal_error(&self) -> bool {
+			matches!(self, InherentError::Fatal)
+		}
+	}
+
+	#[pallet::inherent]
+	impl<T: Config> ProvideInherent for Pallet<T> {
+		type Call = Call<T>;
+		type Error = InherentError;
+
+		const INHERENT_IDENTIFIER: InherentIdentifier = *b"bloated0";
+
+		fn create_inherent(_data: &InherentData) -> Option<Self::Call> {
+			let max_block_length = *T::BlockLength::get().max.get(DispatchClass::Mandatory);
+			let bloat_size = Storage::<T>::get().saturating_mul_int(max_block_length);
```

Comment:
One thing that might be worth changing is adding a separate parameter for block size instead of relying on the `storage` parameter.

Reply:
Using `Storage` is fine. In the end the tx size + the storage proof size = proof size. However, I would not use the `max_block_length` of the mandatory. Maybe only 95% or even better, check the current usage (create_inherent is called after `on_initialize`) and then use 95% of the remaining.

Reply:
Not a good idea, to use same parameter for both with trash data:

```
2024-06-07 16:18:32.422  INFO tokio-runtime-worker aura::cumulus::collation_task: [Parachain] PoV size { header: 0.22kB, extrinsics: 5123.21kB, storage_proof: 1681.66kB }
2024-06-07 16:18:32.423  INFO tokio-runtime-worker aura::cumulus::collation_task: [Parachain] Compressed PoV size: 6739.38671875kb
```

Reply:
This is a bug :D @skunert wanted to fix this :D

Reply:
Yes, that is on the list 👍

Reply:
Is there a ticket ? :D

Reply:
https://github.com/paritytech/polkadot-sdk/issues/4743